modules/gcp/compute.go

```diff
@@ -0,0 +1,196 @@
+	instance, err := service.Instances.Get(projectID, zone, instanceID).Context(ctx).Do()
+	if err != nil {
+		return nil, fmt.Errorf("Instances.Get(%s) got error: %v", instanceID, err)
+	}
+
+	return instance.Labels, nil
+}
+
+// DeleteImage deletes the given Compute Image.
+func DeleteImage(t *testing.T, projectID string, imageID string) {
+	err := DeleteImageE(t, projectID, imageID)
+	if err != nil {
+		t.Fatal(err)
+	}
+}
+
+// DeleteImageE deletes the given Compute Image.
+func DeleteImageE(t *testing.T, projectID string, imageID string) error {
+	logger.Logf(t, "Destroying Image %s", imageID)
+
+	ctx := context.Background()
+
+	service, err := NewComputeServiceE(t)
+	if err != nil {
+		return err
+	}
+
+	if _, err := service.Images.Delete(projectID, imageID).Context(ctx).Do(); err != nil {
+		return fmt.Errorf("Images.Delete(%s) got error: %v", imageID, err)
+	}
+
+	return err
+}
+
+// AddLabelsToInstance adds the tags to the given taggable instance.
+func AddLabelsToInstance(t *testing.T, projectID string, zone string, instance string, labels map[string]string) {
+	err := AddLabelsToInstanceE(t, projectID, zone, instance, labels)
+	if err != nil {
+		t.Fatal(err)
+	}
+}
+
+// AddLabelsToInstanceE adds the tags to the given taggable instance.
+func AddLabelsToInstanceE(t *testing.T, projectID string, zone string, instance string, labels map[string]string) error {
+	logger.Logf(t, "Adding labels to instance %s in zone %s", instance, zone)
+
+	ctx := context.Background()
+
+	service, err := NewComputeServiceE(t)
+	if err != nil {
+		return err
+	}
+
+	// Get the fingerprint of the existing labels
+	existingInstance, err := service.Instances.Get(projectID, zone, instance).Context(ctx).Do()
+	if err != nil {
+		return fmt.Errorf("Instances.Get(%s) got error: %v", instance, err)
+	}
+	req := compute.InstancesSetLabelsRequest{Labels: labels, LabelFingerprint: existingInstance.LabelFingerprint}
+
+	// Perform the SetLabels request
+	if _, err := service.Instances.SetLabels(projectID, zone, instance, &req).Context(ctx).Do(); err != nil {
+		return fmt.Errorf("Instances.SetLabels(%s) got error: %v", instance, err)
+	}
+
+	return err
+}
+
+// GetInstanceIdsForInstanceGroup gets the IDs of Instances in the given Instance Group.
+func GetInstanceIdsForInstanceGroup(t *testing.T, projectID string, zone string, groupName string) []string {
+	ids, err := GetInstanceIdsForInstanceGroupE(t, projectID, zone, groupName)
+	if err != nil {
+		t.Fatal(err)
+	}
+	return ids
+}
+
+// GetInstanceIdsForInstanceGroupE gets the IDs of Instances in the given Instance Group.
+func GetInstanceIdsForInstanceGroupE(t *testing.T, projectID string, zone string, groupName string) ([]string, error) {
+	logger.Logf(t, "Get instances for instance group %s in zone %s", groupName, zone)
+
+	ctx := context.Background()
+
+	service, err := NewComputeServiceE(t)
+	if err != nil {
+		return nil, err
+	}
+
+	rb := &compute.InstanceGroupsListInstancesRequest{
+		InstanceState: "ALL",
+	}
```

Comment:
noob question, but what does `rb` stand for?

Reply:
request body ;), can change if you like?

Reply:
technically its probably in the URL

Reply:
Oh! Haha, wouldn't have guessed request body. Changing to `requestBody` or `req` sounds good to me.

Reply:
ok going with `requestBody`